rustls/src/client/client_conn.rs

```diff
@@ -635,6 +647,12 @@ impl ClientConnection {
         self.inner.core.is_early_data_accepted()
     }
 
+    /// Access connection options whose use is dangerous and requires
+    /// extra care.
+    pub fn dangerous(self) -> danger::DangerousClientConnection {
```

Comment:
I don't love the `.dangerous().extract_secrets()` API. Maybe this should instead be an extension trait that's only exposed via a `danger` module? Like, you have to import `use rustls::danger::SecretExtraction;` in order to call `extract_secrets()` on `ClientConnection` or `ServerConnection`? Alternatively, maybe it would be more general to have the trait be called `DangerousConnection`?

Reply:
> I don't love the .dangerous().extract_secrets() API

Is there something in particular about it you don't like? I sort of like that it's very explicit. An extension trait seems less explicit at the site of the actually dangerous operation.

Reply:
I went with Ctz's suggestion to use `dangerous_extract_secrets`.